internal/logging/logging.go

```diff
@@ -0,0 +1,126 @@
+package logging
+
+import (
+	"context"
+
+	aulogging "github.com/StephanHCB/go-autumn-logging"
+	auzerolog "github.com/StephanHCB/go-autumn-logging-zerolog"
+	auloggingapi "github.com/StephanHCB/go-autumn-logging/api"
+	"github.com/rs/zerolog"
+)
+
+type Logger interface {
+	Debug(format string, v ...interface{})
+	Info(format string, v ...interface{})
+	Warn(format string, v ...interface{})
+	Error(format string, v ...interface{})
+
+	// expected to terminate the process
+	Fatal(format string, v ...interface{})
+}
+
+type loggingWrapper struct {
```

Comment:
Code is copied from the current implementation in the payment service, which if I remember correctly was updated by you to include the go-autumn-logging library.

The wrapper is intended to make calls feel more natural.

Compare:
```go
// autumn code
logger.Info().Printf(format, v...)

// invoked code thanks to logging interface and wrapper
logger.Info(format, v...)

```

Reply:
Matter of taste I guess. I quite like the first format, which is modeled after zerolog's interface.

Reply:
I checked. Back when I introduced go-autumn-logging, I didn't go through the entire code to update every place where the logging Wrapper was called, because the logging wrapper was also passed around as a function argument, and that would have meant bigger code changes. We knew we'd be reworking all the services anyway.

But now that time has come :)

While I don't mind a wrapper if it helps us, autumn logging already includes very similar convenience functions, which will look up the request logger in the context, meaning all log lines will have trace ids set etc.

See https://github.com/StephanHCB/go-autumn-logging/blob/master/convenience.go

We should decide whether we want to use
- a wrapper
- autumn convenience.go callstyle
- autumn object callstyle (same style as zerolog)
and then stick to it across our codebase.

Reply:
https://github.com/eurofurence/reg-room-service/issues/38